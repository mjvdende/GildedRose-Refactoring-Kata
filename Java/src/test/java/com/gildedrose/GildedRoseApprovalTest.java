package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

public class GildedRoseApprovalTest {

    @Test
    public void testApproval(){

        CombinationApprovals.verifyAllCombinations(
                this::doUpdateQuality,
                new String[]{"foo"},
                new Integer[]{0},
                new Integer[]{0});
    }

    private String doUpdateQuality(String name, int sellIn, int quality) {
        Item[] items = new Item[] { new Item(name, sellIn, quality) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }
}
