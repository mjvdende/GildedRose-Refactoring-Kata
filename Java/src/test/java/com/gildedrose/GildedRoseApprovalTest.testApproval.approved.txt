[John, -1, -1] => John, -2, -1 
[John, -1, 0] => John, -2, 0 
[John, -1, 1] => John, -2, 0 
[John, -1, 49] => John, -2, 47 
[John, -1, 50] => John, -2, 48 
[John, 0, -1] => John, -1, -1 
[John, 0, 0] => John, -1, 0 
[John, 0, 1] => John, -1, 0 
[John, 0, 49] => John, -1, 47 
[John, 0, 50] => John, -1, 48 
[John, 6, -1] => John, 5, -1 
[John, 6, 0] => John, 5, 0 
[John, 6, 1] => John, 5, 0 
[John, 6, 49] => John, 5, 48 
[John, 6, 50] => John, 5, 49 
[John, 10, -1] => John, 9, -1 
[John, 10, 0] => John, 9, 0 
[John, 10, 1] => John, 9, 0 
[John, 10, 49] => John, 9, 48 
[John, 10, 50] => John, 9, 49 
[John, 11, -1] => John, 10, -1 
[John, 11, 0] => John, 10, 0 
[John, 11, 1] => John, 10, 0 
[John, 11, 49] => John, 10, 48 
[John, 11, 50] => John, 10, 49 
[John, 49, -1] => John, 48, -1 
[John, 49, 0] => John, 48, 0 
[John, 49, 1] => John, 48, 0 
[John, 49, 49] => John, 48, 48 
[John, 49, 50] => John, 48, 49 
[John, 50, -1] => John, 49, -1 
[John, 50, 0] => John, 49, 0 
[John, 50, 1] => John, 49, 0 
[John, 50, 49] => John, 49, 48 
[John, 50, 50] => John, 49, 49 
[Aged Brie, -1, -1] => Aged Brie, -2, 1 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, 0, -1] => Aged Brie, -1, 1 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 6, -1] => Aged Brie, 5, 0 
[Aged Brie, 6, 0] => Aged Brie, 5, 1 
[Aged Brie, 6, 1] => Aged Brie, 5, 2 
[Aged Brie, 6, 49] => Aged Brie, 5, 50 
[Aged Brie, 6, 50] => Aged Brie, 5, 50 
[Aged Brie, 10, -1] => Aged Brie, 9, 0 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 1] => Aged Brie, 9, 2 
[Aged Brie, 10, 49] => Aged Brie, 9, 50 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 11, -1] => Aged Brie, 10, 0 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 1] => Aged Brie, 10, 2 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Aged Brie, 49, -1] => Aged Brie, 48, 0 
[Aged Brie, 49, 0] => Aged Brie, 48, 1 
[Aged Brie, 49, 1] => Aged Brie, 48, 2 
[Aged Brie, 49, 49] => Aged Brie, 48, 50 
[Aged Brie, 49, 50] => Aged Brie, 48, 50 
[Aged Brie, 50, -1] => Aged Brie, 49, 0 
[Aged Brie, 50, 0] => Aged Brie, 49, 1 
[Aged Brie, 50, 1] => Aged Brie, 49, 2 
[Aged Brie, 50, 49] => Aged Brie, 49, 50 
[Aged Brie, 50, 50] => Aged Brie, 49, 50 
[Backstage passes to a TAFKAL80ETC concert, -1, -1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, -1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 6, -1] => Backstage passes to a TAFKAL80ETC concert, 5, 1 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Backstage passes to a TAFKAL80ETC concert, 5, 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 1] => Backstage passes to a TAFKAL80ETC concert, 5, 3 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, -1] => Backstage passes to a TAFKAL80ETC concert, 9, 1 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Backstage passes to a TAFKAL80ETC concert, 9, 3 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, -1] => Backstage passes to a TAFKAL80ETC concert, 10, 0 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Backstage passes to a TAFKAL80ETC concert, 10, 2 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 49, -1] => Backstage passes to a TAFKAL80ETC concert, 48, 0 
[Backstage passes to a TAFKAL80ETC concert, 49, 0] => Backstage passes to a TAFKAL80ETC concert, 48, 1 
[Backstage passes to a TAFKAL80ETC concert, 49, 1] => Backstage passes to a TAFKAL80ETC concert, 48, 2 
[Backstage passes to a TAFKAL80ETC concert, 49, 49] => Backstage passes to a TAFKAL80ETC concert, 48, 50 
[Backstage passes to a TAFKAL80ETC concert, 49, 50] => Backstage passes to a TAFKAL80ETC concert, 48, 50 
[Backstage passes to a TAFKAL80ETC concert, 50, -1] => Backstage passes to a TAFKAL80ETC concert, 49, 0 
[Backstage passes to a TAFKAL80ETC concert, 50, 0] => Backstage passes to a TAFKAL80ETC concert, 49, 1 
[Backstage passes to a TAFKAL80ETC concert, 50, 1] => Backstage passes to a TAFKAL80ETC concert, 49, 2 
[Backstage passes to a TAFKAL80ETC concert, 50, 49] => Backstage passes to a TAFKAL80ETC concert, 49, 50 
[Backstage passes to a TAFKAL80ETC concert, 50, 50] => Backstage passes to a TAFKAL80ETC concert, 49, 50 
[Sulfuras, Hand of Ragnaros, -1, -1] => Sulfuras, Hand of Ragnaros, -1, -1 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, 0, -1] => Sulfuras, Hand of Ragnaros, 0, -1 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 6, -1] => Sulfuras, Hand of Ragnaros, 6, -1 
[Sulfuras, Hand of Ragnaros, 6, 0] => Sulfuras, Hand of Ragnaros, 6, 0 
[Sulfuras, Hand of Ragnaros, 6, 1] => Sulfuras, Hand of Ragnaros, 6, 1 
[Sulfuras, Hand of Ragnaros, 6, 49] => Sulfuras, Hand of Ragnaros, 6, 49 
[Sulfuras, Hand of Ragnaros, 6, 50] => Sulfuras, Hand of Ragnaros, 6, 50 
[Sulfuras, Hand of Ragnaros, 10, -1] => Sulfuras, Hand of Ragnaros, 10, -1 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[Sulfuras, Hand of Ragnaros, 10, 1] => Sulfuras, Hand of Ragnaros, 10, 1 
[Sulfuras, Hand of Ragnaros, 10, 49] => Sulfuras, Hand of Ragnaros, 10, 49 
[Sulfuras, Hand of Ragnaros, 10, 50] => Sulfuras, Hand of Ragnaros, 10, 50 
[Sulfuras, Hand of Ragnaros, 11, -1] => Sulfuras, Hand of Ragnaros, 11, -1 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 1] => Sulfuras, Hand of Ragnaros, 11, 1 
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
[Sulfuras, Hand of Ragnaros, 49, -1] => Sulfuras, Hand of Ragnaros, 49, -1 
[Sulfuras, Hand of Ragnaros, 49, 0] => Sulfuras, Hand of Ragnaros, 49, 0 
[Sulfuras, Hand of Ragnaros, 49, 1] => Sulfuras, Hand of Ragnaros, 49, 1 
[Sulfuras, Hand of Ragnaros, 49, 49] => Sulfuras, Hand of Ragnaros, 49, 49 
[Sulfuras, Hand of Ragnaros, 49, 50] => Sulfuras, Hand of Ragnaros, 49, 50 
[Sulfuras, Hand of Ragnaros, 50, -1] => Sulfuras, Hand of Ragnaros, 50, -1 
[Sulfuras, Hand of Ragnaros, 50, 0] => Sulfuras, Hand of Ragnaros, 50, 0 
[Sulfuras, Hand of Ragnaros, 50, 1] => Sulfuras, Hand of Ragnaros, 50, 1 
[Sulfuras, Hand of Ragnaros, 50, 49] => Sulfuras, Hand of Ragnaros, 50, 49 
[Sulfuras, Hand of Ragnaros, 50, 50] => Sulfuras, Hand of Ragnaros, 50, 50 
